### DISCUSSION TITLE
MODELS folder empty - data corruption after lua script

### DISCUSSION THREAD

**Discussion by RobertoD91** [EdgeTX/edgetx#5229] [Q&A]
I have an jumper t16 with edgetx 2.10.1
I just discover that many models was missing, ~~if i connect the sd to the computer the folder MODELS is empty.~~

Any idea on that is happened?

EDIT:
<img width="570" alt="image" src="https://github.com/EdgeTX/edgetx/assets/298514/760d4694-9dbb-43b3-9118-85549fa67fbf">

Connected to a linux vm, the MODELS folder is a mess...

According to https://github.com/EdgeTX/edgetx/issues/5077 his may be caused by some DSM lua script?

Any idea for recovering the models?

**Comment by pfeerick** [comment]
Unfortunately no. If you don't have any backups or etx file saved from Companion, you're sunk, as the card is corrupt.

**Comment by RobertoD91** [comment]
The Windows Disk Repair utility fixed the SD card, and some .yml files from the models reappeared in a "found" folder.
The files were corrupted.

By using a backup, some tools to compare file differences, and a lot of manual work, I managed to recover almost everything.

However, such a serious bug should probably be reported to users, so they can make a backup before running any Lua scripts.

**Reply by pfeerick** [reply]
Glad to hear it. Yeah, chkdsk would be pretty unlikely to give any useful results since the files are being partially over-written, not just missing or deleted.

It was mentioned when 2.10.1 was released, with a subsequent update when more information was known. 
> We are aware of reports of issues related to SD cards not working with 2.10 and corruption issues for some people and are investigating the cause/resolution for that, so stay tuned for an update on that. edit: It appears they are related to Lua scripts with a lot of SD card usage, not just general usage.

v2.10.2 *should* resolve this. 
> Regarding the previously mentioned SD card corruption issues that some people were experiencing, it appears they are related to Lua scripts with a lot of SD card usage, not just general usage. One such example was the DSM Forward Programming Lua tool. There is a fix in place that should reduce if not completely resolve this, and we will continue to investigate and monitor the situat

**Reply by RobertoD91** [reply]
> Glad to hear it. Yeah, chkdsk would be pretty unlikely to give any useful results since the files are being partially over-written, not just missing or deleted.

Yes, it was a partial recovery, but it was still useful. I also tried some recovery tools (photorec and others) but the integrated Windows tool surprised me.

> v2.10.2 _should_ resolve this.

Thanks for your work! ❤️
